.github/workflows/testing.yml

```diff
@@ -9,116 +9,69 @@ on:
-            os: ["ubuntu-latest", "macos-latest", "windows-latest"]
-            python-version: ["3.9", "3.10", "3.11", "3.12"]
-            # Install the min or latest dependencies for skrub
-            # as defined in setup.cfg at [options.extra_require].
-            #
-            # Installing these extra dependencies will overwrite the already
-            # installed version defined at install_requires.
-            dependencies-version-type: ["minimal", "latest"]
-            include:
-              - os: "ubuntu-latest"
-                os-name: "Ubuntu"
-              - os: "macos-latest"
-                os-name: "MacOS"
-              - os: "windows-latest"
-                os-name: "Windows"
-              - dependencies-version: "dev"
-              - dependencies-version: "dev, pyarrow"
-                python-version: "3.11"
-              - dependencies-version: "dev, polars"
-                python-version: "3.11"
-              - dependencies-version: "dev, polars"
-                python-version: "3.12"
-              - dependencies-version: "dev, min-py39"
-                python-version: "3.10"
-                dependencies-version-type: "minimal"
-              - dependencies-version: "dev, min-py39"
-                python-version: "3.9"
-                dependencies-version-type: "minimal"
-    name: ${{ matrix.os-name }} with Python ${{ matrix.python-version }} and ${{ matrix.dependencies-version-type }} dependencies
-    defaults:
-      run:
-        shell: bash
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        environment: [
+            ci-py309-min-deps,
+            ci-py310-min-optional-deps,
+            ci-py312-latest-deps,
+            ci-py312-latest-optional-deps
+        ]
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
+      - uses: prefix-dev/setup-pixi@v0.8.1
         with:
-            python-version: ${{ matrix.python-version }}
-        name: 'Setup python'
-      - shell: bash {0}
-        run: ./build_tools/github/install.sh
-        name: 'Install skrub'
-        env:
-          DEPS_VERSION: ${{ matrix.dependencies-version }}
-      - shell: bash {0}
-        run: |
-          cp $GITHUB_WORKSPACE/pyproject.toml .
-          $GITHUB_WORKSPACE/build_tools/github/test.sh
-        working-directory: ${{ runner.temp }}
-        name: 'Run tests'
-      - uses: codecov/codecov-action@v3
+          pixi-version: v0.24.0
+          environments: ${{ matrix.environment }}
+          # we can freeze the environment and manually bump the dependencies to the
+          # latest version time to time.
+          frozen: true
+
+      - name: Run tests
+        run: pixi run -e ${{ matrix.environment }} test -n 3
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.4.1
         with:
-          directory: ${{ runner.temp }}
```

Comment:
does this mean we are now running the tests from the source directory? I think that's not ideal with the current layout of the package because it could end up being imported from the pwd? or maybe I am missing something